src/main/java/com/rultor/profiles/Profiles.java

```diff
@@ -73,18 +79,77 @@ public Profile fetch(final Talk talk) throws IOException {
             if (xml.nodes("/talk/wire").isEmpty()) {
                 profile = Profile.EMPTY;
             } else {
-                profile = Profiles.fetch(xml);
+                profile = this.fetch(xml);
             }
         }
         return profile;
     }
 
     /**
+     * Merge profile from master and fork. Merged profile must be profile
+     * from fork, but all lists for commanders and architects must be taken
+     * from master branch.
+     * @param master Profile from master branch
+     * @param fork Profile from fork branch
+     * @return Merged profile
+     * @todo #1039:30min Modify method implementation: merged profile is a
+     *  `.rultor.yml` form result of real `git merge` of master branch and
+     *  fork branch. We need real `git merge` result here. See some details
+     *  here: https://github.com/yegor256/rultor/pull/1064/files#r56796959
+     */
+    public Profile merged(final Profile master, final Profile fork) {
+        return master;
+    }
+
+    /**
+     * Validate merged profile: merged profile must have no changes in architect
+     * and commander sections (full match with master profile).
+     * @param master Profile from master branch
+     * @param merged Result of merging master and fork profiles
+     * @return Validated profile
+     * @throws IOException If fails
+     */
+    public Profile validated(final Profile master, final Profile merged)
+        throws IOException {
+        final String[] security = {"architect", "commander"};
+        for (final String section : security) {
+            if (!Arrays.equals(
+                section(master, section),
+                section(merged, section)
+            )) {
+                throw new Profile.ConfigException(
```

Comment:
@xupyprmv oh no that's something else, that's the thing that generates this green dotted timeline on http://rultor.com :)

I suggest to simply remove the call to this function for now then, add to the puzzle that this should be used on the actual merge result eventually ( it seems to work nicely looking at the tests! :) ), but don't use it as of now in production code. Not worth adding a random risk here, it's simply not properly tested at this point and a little tricky to test too, you've done enough here no need for you to add the IT case for that. Just adjust puzzle and remove the call to this for now :)
Sounds reasonable ?